### ISSUE TITLE
tarpaulin does not compile currently

### ISSUE THREAD

**Issue by vberger** [xd009642/tarpaulin#76]
All my CI fails with the same error, both stable, beta and nighlty.

Here is the output of `cargo install cargo-tarpaulin`:

```
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading cargo-tarpaulin v0.5.6
  Installing cargo-tarpaulin v0.5.6
   Compiling foreign-types-shared v0.1.1
   Compiling scroll v0.5.0
   Compiling matches v0.1.6
   Compiling unicode-width v0.1.4
   Compiling regex-syntax v0.4.2
   Compiling unicode-normalization v0.1.5
   Compiling hex v0.2.0
   Compiling openssl v0.9.23
   Compiling num-traits v0.1.41
   Compiling plain v0.0.2
   Compiling crossbeam v0.2.10
   Compiling term v0.4.6
   Compiling libc v0.2.34
   Compiling fnv v1.0.6
   Compiling void v1.0.2
   Compiling vec_map v0.8.0
   Compiling bitflags v0.9.1
   Compiling unicode-xid v0.0.4
   Compiling shell-escape v0.1.3
   Compiling glob v0.2.11
   Compiling home v0.3.0
   Compiling unicode-xid v0.1.0
   Compiling strsim v0.6.0
   Compiling bitflags v1.0.1
   Compiling quote v0.3.15
   Compiling utf8-ranges v1.0.0
   Compiling scoped-tls v0.1.0
   Compiling same-file v0.1.3
   Compiling itoa v0.3.4
   Compiling pkg-config v0.3.9
   Compiling percent-encoding v1.0.1
   Compiling build_const v0.2.0
   Compiling lazy_static v1.0.0
   Compiling dtoa v0.4.2
   Compiling md5 v0.3.6
   Compiling cc v1.0.3
   Compiling rustc-demangle v0.1.5
   Compiling ansi_term v0.10.2
   Compiling semver-parser v0.7.0
   Compiling odds v0.2.26
   Compiling nodrop v0.1.12
   Compiling openssl-probe v0.1.2
   Compiling adler32 v1.0.2
   Compiling byteorder v1.2.1
   Compiling termcolor v0.3.3
   Compiling fallible-iterator v0.1.4
   Compiling cfg-if v0.1.2
   Compiling lazy_static v0.2.11
   Compiling foreign-types v0.3.2
   Compiling textwrap v0.9.0
   Compiling unicode-bidi v0.3.4
   Compiling enum_primitive v0.1.1
   Compiling atty v0.2.6
   Compiling memchr v1.0.2
   Compiling jobserver v0.1.8
   Compiling fs2 v0.4.2
   Compiling memchr v2.0.1
   Compiling memmap v0.5.2
   Compiling num_cpus v1.8.0
   Compiling rand v0.3.19
   Compiling unreachable v1.0.0
   Compiling synom v0.11.3
   Compiling walkdir v1.0.7
   Compiling crc v1.7.0
   Compiling arrayvec v0.3.24
   Compiling semver v0.6.0
   Compiling filetime v0.1.14
   Compiling encoding_rs v0.7.1
   Compiling log v0.4.1
   Compiling socket2 v0.2.4
   Compiling nix v0.9.0
   Compiling clap v2.29.0
   Compiling miniz-sys v0.1.10
   Compiling libz-sys v1.0.18
   Compiling backtrace-sys v0.1.16
   Compiling cmake v0.1.29
   Compiling openssl-sys v0.9.23
   Compiling curl-sys v0.3.15
   Compiling aho-corasick v0.6.4
   Compiling idna v0.1.4
   Compiling thread_local v0.3.5
   Compiling syn v0.11.11
   Compiling gimli v0.14.0
   Compiling tar v0.4.14
   Compiling tempdir v0.3.5
   Compiling log v0.3.9
   Compiling rustc_version v0.2.1
   Compiling libssh2-sys v0.2.6
   Compiling libgit2-sys v0.6.18
   Compiling regex v0.2.5
   Compiling url v1.6.0
   Compiling extprim v1.4.0
   Compiling serde_derive_internals v0.19.0
   Compiling scroll_derive v0.4.0
   Compiling gzip-header v0.1.2
   Compiling flate2 v0.2.20
   Compiling goblin v0.0.10
   Compiling serde_derive v1.0.27
   Compiling backtrace v0.3.4
   Compiling deflate v0.7.17
   Compiling globset v0.2.1
   Compiling env_logger v0.4.3
   Compiling curl v0.4.8
   Compiling object v0.4.1
   Compiling error-chain v0.11.0
   Compiling error-chain v0.10.0
   Compiling ignore v0.2.2
   Compiling git2 v0.6.10
   Compiling quick-xml v0.9.4
   Compiling git2-curl v0.7.0
   Compiling serde v1.0.27
   Compiling serde_json v1.0.9
   Compiling syntex_pos v0.59.1
   Compiling serde_ignored v0.0.3
   Compiling toml v0.4.5
   Compiling docopt v0.8.2
   Compiling semver v0.7.0
   Compiling syntex_errors v0.59.1
   Compiling coveralls-api v0.3.3
   Compiling crates-io v0.11.0
   Compiling syntex_syntax v0.59.1
   Compiling cargo v0.22.0
error[E0277]: the trait bound `for<'de> Flags: serde::Deserialize<'de>` is not satisfied
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/cargo-0.22.0/src/cargo/lib.rs:123:24
    |
123 |     let flags = docopt.deserialize().map_err(|e| {
    |                        ^^^^^^^^^^^ the trait `for<'de> serde::Deserialize<'de>` is not implemented for `Flags`
    |
    = help: consider adding a `where for<'de> Flags: serde::Deserialize<'de>` bound
    = note: required because of the requirements on the impl of `serde::de::DeserializeOwned` for `Flags`

error: aborting due to previous error

error: Could not compile `cargo`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-tarpaulin v0.5.6`, intermediate artifacts can be found at `/tmp/cargo-install.LR3HIPbaHUmt`
```

**Comment by xd009642** [comment]
Well the compile error is in cargo itself which is mighty strange. Have you got all the dependencies installed?

**Comment by vberger** [comment]
Okay, problem is resolved, there was a breaking change in `docopt-0.8.2` that was fixed in `docopt-0.8.3` : https://github.com/docopt/docopt.rs/commit/7292a374e69afb192bb7aaa00f9d9f4afebc200d
